lib/stages/CMakeLists.txt

```diff
@@ -1,131 +1,115 @@
+    rfiAVXVDIF.cpp
+    rfiBroadcast.cpp
+    rfiRecord.cpp
+    rfiBadInputFinder.cpp
+    rfiUpdateMetadata.cpp
+    ReceiveFlags.cpp
+    valve.cpp
+    visTransform.cpp
+    visTestPattern.cpp
+    visDebug.cpp
+    removeEv.cpp
+    integrateHFBData.cpp
+    compressLostSamples.cpp
+    compressData.cpp
+    bufferBadInputs.cpp
+    RfiFrameDrop.cpp
+    VisSharedMemWriter.cpp)
 
-set ( KOTEKAN_PROCESS_LIB_SOURCES
-      beamformingPostProcess.cpp
-      chrxUplink.cpp
-      computeDualpolPower.cpp
-      fullPacketDump.cpp
-      pulsarPostProcess.cpp
-      ReadGain.cpp
-      nDiskFileWrite.cpp
-      nDiskFileRead.cpp
-      networkPowerStream.cpp
-      integratePowerStream.cpp
-      monitorBuffer.cpp
-      networkInputPowerStream.cpp
-      pyPlotN2.cpp
-      rawFileRead.cpp
-      rawFileWrite.cpp
-      vdifStream.cpp
-      recvSingleDishVDIF.cpp
-      streamSingleDishVDIF.cpp
-      accumulate.cpp
-      bufferStatus.cpp
-      bufferSend.cpp
-      bufferRecv.cpp
-      simpleAutocorr.cpp
-      freqSplit.cpp
-      freqSubset.cpp
-      prodSubset.cpp
-      countCheck.cpp
-      visAccumulate.cpp
-      visCompression.cpp
-      timeDownsample.cpp
-      frbPostProcess.cpp
-      visWriter.cpp
-      visRawReader.cpp
-      restInspectFrame.cpp
-      frbNetworkProcess.cpp
-      pulsarNetworkProcess.cpp
-      bufferMerge.cpp
-      bufferCopy.cpp
-      bufferSwitch.cpp
-      bufferDelay.cpp
-      InputSubset.cpp
-      #RFI Pipeline Processes
-      rfiVDIF.cpp
-      rfiAVXVDIF.cpp
-      rfiBroadcast.cpp
-      rfiRecord.cpp
-      rfiBadInputFinder.cpp
-      rfiUpdateMetadata.cpp
-      ReceiveFlags.cpp
-      valve.cpp
-      visTransform.cpp
-      visTestPattern.cpp
-      visDebug.cpp
-      removeEv.cpp
-      integrateHFBData.cpp
-      compressLostSamples.cpp
-      compressData.cpp
-      bufferBadInputs.cpp
-      RfiFrameDrop.cpp
-      VisSharedMemWriter.cpp
-    )
+target_include_directories(kotekan_stages PRIVATE .)
 
+if(${USE_AIRSPY})
+    target_sources(kotekan_stages PRIVATE airspyInput.cpp)
+    target_include_directories(kotekan_stages SYSTEM PRIVATE ${LIBAIRSPY_INCLUDE_DIR})
+    target_include_directories(kotekan_stages SYSTEM PRIVATE ${FFTW_INCLUDES})
+    target_link_libraries(kotekan_stages PRIVATE ${LIBAIRSPY_LIBRARIES})
+endif()
 
-if (${USE_AIRSPY})
-  include_directories(${LIBAIRSPY_INCLUDE_DIR})
-  set (KOTEKAN_PROCESS_LIB_SOURCES ${KOTEKAN_PROCESS_LIB_SOURCES} airspyInput.cpp)
-endif ()
-if (${USE_FFTW})
-  set (KOTEKAN_PROCESS_LIB_SOURCES ${KOTEKAN_PROCESS_LIB_SOURCES} fftwEngine.cpp)
-endif ()
-
-# HDF5 stuff
-if (${USE_HDF5})
-  set (KOTEKAN_PROCESS_LIB_SOURCES ${KOTEKAN_PROCESS_LIB_SOURCES} basebandReadout.cpp applyGains.cpp)
-endif ()
-
-# HDF5 and OpenMP
-if (${USE_HDF5} AND ${USE_OMP})
-  set (KOTEKAN_PROCESS_LIB_SOURCES ${KOTEKAN_PROCESS_LIB_SOURCES} visTranspose.cpp)
+if(${USE_FFTW})
+    set(KOTEKAN_PROCESS_LIB_SOURCES ${KOTEKAN_PROCESS_LIB_SOURCES} fftwEngine.cpp)
```

Comment:
It seems odd that this one still needs an explicit `set`. Is that correct? If so, maybe add a comment about it.

Reply:
Looks like I forgot this one